src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -16,16 +16,39 @@
 
 using System.Diagnostics.Metrics;
 using System.Reflection;
+using Diagnostics = System.Diagnostics;
 
 namespace OpenTelemetry.Instrumentation.Process;
 
 internal class ProcessMetrics
 {
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
     internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
+    private static readonly Diagnostics.Process CurrentProcess = Diagnostics.Process.GetCurrentProcess();
```

Comment:
This, combined with `Refresh`, look like a source of race condition.

![image](https://user-images.githubusercontent.com/17327289/186954796-0f56183f-2441-4556-b903-9094e5d7912d.png)

Reply:
@Yun-Ting You could probably just use a closure a la:

```csharp
    static ProcessMetrics()
    {
        Diagnostics.Process currentProcess = Diagnostics.Process.GetCurrentProcess();

        // TODO: change to ObservableUpDownCounter
        MeterInstance.CreateObservableGauge(
            "process.memory.usage",
            () =>
            {
                currentProcess.Refresh();
                return currentProcess.WorkingSet64;
            },
            unit: "By",
            description: "The amount of physical memory in use.");

        // TODO: change to ObservableUpDownCounter
        MeterInstance.CreateObservableGauge(
            "process.memory.virtual",
            () =>
            {
                currentProcess.Refresh();
                return currentProcess.VirtualMemorySize64;
            },
            unit: "By",
            description: "The amount of committed virtual memory.");
    }
```

Is `Process.Refresh()` expensive? We'll be invoking it twice here anytime metrics are collected. If it is expensive we might want to create a state object we can refresh once and then return the result to the two counters in the callback.

Reply:
Thank you very much for pointing it out. Please refer to https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/612 for the update.

Reply:
Please refer to the latest PR https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/624 instead. Thanks!